File: 프로그래머스/2/77485. 행렬 테두리 회전하기/행렬 테두리 회전하기.js
function solution(rows, columns, queries) {
    let answer = [];
    
    let board = Array.from(Array(rows + 1), () => new Array(columns + 1));
    let num = 1;
    for(let r = 1; r <= rows; r++) {
        for(let c = 1; c <= columns; c++) {
            board[r][c] = num++;
        }
    }
    
    function TurnNum(startR, startC, endR, endC) {
        let minNum = board[startR][startC], tmp = board[startR][startC];
    
        // 왼 |
        for(let r = startR; r < endR; r++) {
            minNum = Math.min(minNum, board[r][startC]);
            board[r][startC] = board[r + 1][startC];
        }
        // 아래 ㅡ
        for(let c = startC; c < endC; c++) {
            minNum = Math.min(minNum, board[endR][c]);
            board[endR][c] = board[endR][c + 1];
        }
        // 오 |
        for(let r = endR; r > startR; r--) {
            minNum = Math.min(minNum, board[r][endC]);
            board[r][endC] = board[r - 1][endC];
        }
        // 위 ㅡ
        for(let c = endC; c > startC; c--) {
            minNum = Math.min(minNum, board[startR][c]);
            board[startR][c] = board[startR][c - 1];
        }

        board[startR][startC + 1] = tmp;
        return minNum;
    }
    
    for(let i = 0; i < queries.length; i++) {
        answer.push(TurnNum(queries[i][0], queries[i][1], queries[i][2], queries[i][3]));
    } 
    
    return answer;
}

File: 프로그래머스/2/77485. 행렬 테두리 회전하기/행렬 테두리 회전하기.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int TurnNum(vector<vector<int>> &board, int startR, int startC, int endR, int endC) {
    int minNum = board[startR][startC], tmp = board[startR][startC];
    
    // 왼 |
    for(int r = startR; r < endR; r++) {
        minNum = min(minNum, board[r][startC]);
        board[r][startC] = board[r + 1][startC];
    }
    // 아래 ㅡ
    for(int c = startC; c < endC; c++) {
        minNum = min(minNum, board[endR][c]);
        board[endR][c] = board[endR][c + 1];
    }
    // 오 |
    for(int r = endR; r > startR; r--) {
        minNum = min(minNum, board[r][endC]);
        board[r][endC] = board[r - 1][endC];
    }
    // 위 ㅡ
    for(int c = endC; c > startC; c--) {
        minNum = min(minNum, board[startR][c]);
        board[startR][c] = board[startR][c - 1];
    }
    
    board[startR][startC + 1] = tmp;
    return minNum;
}

vector<int> solution(int rows, int columns, vector<vector<int>> queries) {
    vector<int> answer;
    
    int num = 1;
    vector<vector<int>> board(rows + 1, vector<int> (columns + 1));
    for(int i = 1; i <= rows; i++) {
        for(int j = 1; j <= columns; j++) {
            board[i][j] = num++;
        }
    }
    
    int cur = 0, tmp = 0;
    for(int i = 0; i < queries.size(); i++)
        answer.push_back(TurnNum(board, queries[i][0], queries[i][1], queries[i][2], queries[i][3]));
    
    return answer;
}